cmd/vcert/cmdCloudKeystores.go

```diff
@@ -76,19 +76,24 @@ func doCommandProvisionCloudKeystore(c *cli.Context) error {
 	}
 
 	result := ProvisioningResult{
-		ARN:          metadata.GetAWSCertificateMetadata().GetARN(),
-		AzureID:      metadata.GetAzureCertificateMetadata().GetID(),
-		AzureName:    metadata.GetAzureCertificateMetadata().GetName(),
-		AzureVersion: metadata.GetAzureCertificateMetadata().GetVersion(),
-		GcpID:        metadata.GetGCPCertificateMetadata().GetID(),
-		GcpName:      metadata.GetGCPCertificateMetadata().GetName(),
+		MachineIdentityId:         metadata.MachineIdentityID,
+		MachineIdentityActionType: metadata.MachineIdentityActionType,
+	}
+	switch metadata.CloudKeystoreType {
+	case domain.CloudKeystoreTypeACM:
+		result.ARN = metadata.ARN
+	case domain.CloudKeystoreTypeAKV:
+		result.AzureID = metadata.CertificateID
+		result.AzureName = metadata.CertificateName
+		result.AzureVersion = metadata.CertificateVersion
+	case domain.CloudKeystoreTypeGCM:
+		result.GcpID = metadata.CertificateID
+		result.GcpName = metadata.CertificateName
+	default:
+		return fmt.Errorf("unknown keystore metadata type: %s", metadata.CloudKeystoreType)
 	}
 
-	result.MachineIdentityId = metadata.GetMachineIdentityMetadata().GetID()
-	result.MachineIdentityActionType = metadata.GetMachineIdentityMetadata().GetActionType()
-
-	err = result.Flush(flags.provisionFormat, flags.provisionOutputFile)
```

Comment:
Why did you remove this? we need this for --file flag to work

Reply:
This got lost during branch rebase. Added back.